kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -0,0 +1,224 @@
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+        self._table_name = table_name
+        self._database = database
+        self._schema = schema
+
+        connection_parameters = credentials
+        connection_parameters.update(
+            {"database": self._database, "schema": self._schema}
+        )
+        self._connection_parameters = connection_parameters
+        self._session = self._get_session(self._connection_parameters)
+
+    def _describe(self) -> Dict[str, Any]:
+        return {
+            "table_name": self._table_name,
+            "database": self._database,
+            "schema": self._schema,
+        }
+
+    @staticmethod
+    def _get_session(connection_parameters) -> sp.Session:
+        """Given a connection string, create singleton connection
+        to be used across all instances of `SnowparkDataSet` that
+        need to connect to the same source.
+        connection_parameters is a dictionary of any values
+        supported by snowflake python connector:
+                https://docs.snowflake.com/en/user-guide/python-connector-api.html#connect
+            example:
+            connection_parameters = {
+                "account": "",
+                "user": "",
+                "password": "", (optional)
+                "role": "", (optional)
+                "warehouse": "", (optional)
+                "database": "", (optional)
+                "schema": "", (optional)
+                "authenticator: "" (optional)
+                }
+        """
+        try:
+            logger.debug("Trying to reuse active snowpark session...")
+            # if hook is implemented, get active session
+            session = sp.context.get_active_session()
+        except sp.exceptions.SnowparkSessionException:
+            # create session if there is no active one
+            logger.debug("No active snowpark session found. Creating")
+            session = sp.Session.builder.configs(connection_parameters).create()
+        return session
+
+    def _load(self) -> sp.DataFrame:
+        table_name = [
+            self._database,
+            self._schema,
+            self._table_name,
+        ]
+
+        sp_df = self._session.table(".".join(table_name))
+        return sp_df
+
+    def _save(self, data: Union[pd.DataFrame, sp.DataFrame]) -> None:
+        if not isinstance(data, sp.DataFrame):
+            sp_df = self._session.create_dataframe(data)
+        else:
+            sp_df = data
+
+        table_name = [
+            self._database,
+            self._schema,
+            self._table_name,
+        ]
+
+        sp_df.write.save_as_table(table_name, **self._save_args)
+
+    def _exists(self) -> bool:
+        session = self._session
+        query = "SELECT COUNT(*) FROM {database}.INFORMATION_SCHEMA.TABLES \
```

Comment:
But this part won't apply for queries

Reply:
As current scope is only tables/views we should be fine

Reply:
I think like the SparkDataSet it makes sense to limit this to just tables in the first instance